### ISSUE TITLE
Include libunity to get badges in dash-to-dock

### ISSUE THREAD

**Issue by gicmo** [flathub/org.gnome.Geary#13]
For users of dash-to-dock it would be nice to include libunity so we get badges on the dock for new mail.

**Comment by mjog** [comment]
Hey, thanks for the request.

DO you know if libunity is actually needed by dash-t-dock, or if just libmessagingmenu after Ubuntu switched back to GNOME?

**Comment by mjog** [comment]
Ah, nevermind, worked it out myself.

**Comment by mjog** [comment]
Can you try installing this test build and let me know if it works for you?

```
flatpak install https://dl.flathub.org/build-repo/259/org.gnome.Geary.flatpakref
```

Ta!

**Comment by mjog** [comment]
^-- @gicmo

**Comment by gicmo** [comment]
Thanks! I tried it and it does not work :( I had a look and the generated `config.h` looks correct:

```
#undef HAVE_LIBMESSAGINGMENU

#define HAVE_LIBUNITY
```
Also the compile command seems ok, i.e. including the `-include config.h`:
```
cc -Isrc/client/f537023@@geary-client@sta -Isrc/client -I../src/client -I. -I../ -Isrc/engine -I../src/engine -Iui -I/usr/include/enchant-2 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/gck-1 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid -I/usr/include/p11-kit-1 -I/usr/include/gcr-3 -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/libdrm -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gee-0.8 -I/usr/include/goa-1.0 -I/usr/lib64/goa-1.0/include -I/usr/include/json-glib-1.0 -I/usr/include/libsecret-1 -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/usr/include/gmime-2.6 -I/usr/include/webkitgtk-4.0 -I/usr/include/unity/unity -I/usr/include/dee-1.0 -I/usr/include/libdbusmenu-glib-0.4 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -w -g -fPIC -pthread -D_LARGEFILE64_SOURCE -D_REENTRANT -include config.h -DUNW_LOCAL_ONLY -DGOA_API_IS_SUBJECT_TO_CHANGE -DGCK_API_SUBJECT_TO_CHANGE  -MD -MQ 'src/client/f537023@@geary-client@sta/meson-generated_notification_unity-launcher.c.o' -MF 'src/client/f537023@@geary-client@sta/meson-generated_notification_unity-launcher.c.o.d' -o 'src/client/f537023@@geary-client@sta/meson-generated_notification_unity-launcher.c.o' -c 'src/client/f537023@@geary-client@sta/notification/unity-launcher.c'
```

But somehow the generated code is as if the `HAVE_LIBUNITY` is not present (I added a `debug (FAILED TO ...);` statement:
```
UnityLauncher*
unity_launcher_construct (GType object_type,
                          NewMessagesMonitor* monitor)
{
	UnityLauncher * self = NULL;
#line 56 "../src/client/notification/unity-launcher.vala"
	g_return_val_if_fail (IS_NEW_MESSAGES_MONITOR (monitor), NULL);
#line 56 "../src/client/notification/unity-launcher.vala"
	self = (UnityLauncher*) geary_base_object_construct (object_type);
#line 57 "../src/client/notification/unity-launcher.vala"
	g_debug ("unity-launcher.vala:57: FAILED TO CREATE UNITY LAUNCHER");
#line 56 "../src/client/notification/unity-launcher.vala"
	return self;
#line 75 "unity-launcher.c"
}
```
Is `config.h` too late for the vala compiler to pick up the (vala) preprocessor defines?
As a quick hack I did add the following to the main `meson.build` and that *did* work and made the badges work:

```
if libunity.found()
  add_project_arguments('--define=HAVE_LIBUNITY', language : 'vala')
endif
```

**Comment by mjog** [comment]
Upstream bug: https://gitlab.gnome.org/GNOME/geary/issues/264

Will be fixed in 0.13.2
